### PR TITLE
[new release] lzo (0.0.3)

### PR DESCRIPTION
**Pull request by ygrek**
Bindings to LZO - a portable lossless data compression library

- Project page: <a href="https://ygrek.org/p/ocaml-lzo/">https://ygrek.org/p/ocaml-lzo/</a>
- Documentation: <a href="https://ygrek.org/p/ocaml-lzo/api/index.html">https://ygrek.org/p/ocaml-lzo/api/index.html</a>

#####

### CODE REVIEW

**Comment by raphael-proust** [comment]
There's a warning about a missing license. What are the bindings licensed as?

**Comment by raphael-proust** [comment]
None of the builds pass. The error seem to be with a missing library:

```
#=== ERROR while compiling lzo.0.0.3 ==========================================#
# context              2.2.0~beta3~dev | linux/x86_64 | ocaml-base-compiler.5.2.0 | pinned(https://github.com/ygrek/ocaml-lzo/releases/download/v0.0.3/lzo-0.0.3.tbz)
# path                 ~/.opam/5.2/.opam-switch/build/lzo.0.0.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p lzo -j 31
# exit-code            1
# env-file             ~/.opam/log/lzo-7-80e868.env
# output-file          ~/.opam/log/lzo-7-80e868.out
### output ###
# File "src/dune", line 7, characters 9-18:
# 7 |   (names lzo_stubs)
#              ^^^^^^^^^
# (cd _build/default/src && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -pthread -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -pthread -g -I /home/opam/.opam/5.2/lib/ocaml -o lzo_stubs.o -c lzo_stubs.c)
# lzo_stubs.c:2:10: fatal error: lzodefs.h: No such file or directory
#     2 | #include <lzodefs.h>
#       |          ^~~~~~~~~~~
# compilation terminated.
```

I think that you need to add a `conf-lzo` package and depend on it from your bindings package. To populate the system package names in the conf package you can check: https://pkgs.org/search/?q=lzo

**Comment by ygrek** [comment]
there are depexts in the package and can see from ci log the c library is installed
i suspect it is pkg-config missing and the build process doesn't work without pkg-config @XVilka

File: packages/lzo/lzo.0.0.3/opam [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "conf-pkg-config"
+]
+depexts: [
+  ["lzo2"] {os-family = "bsd"}
+  ["liblzo2-dev"] {os-family = "debian"}
+  ["lzo"] {os-distribution = "homebrew"}
+  ["lzo-dev"] {os-distribution = "alpine"}
+  ["lzo-devel"] {os-family = "suse" | os-family = "opensuse"}
+  ["lzo-devel"] {os-distribution = "centos" | os-distribution = "fedora"}
+]
```
Comment (raphael-proust) [inline packages/lzo/lzo.0.0.3/opam:26]:
```suggestion
  ["lzo"] {os-family = "archlinux"}
  ["lzo-devel"] {os-distribution = "ol"}
]
```

**Comment by XVilka** [comment]
> there are depexts in the package and can see from ci log the c library is installed i suspect it is pkg-config missing and the build process doesn't work without pkg-config @XVilka

Yes, it depends on the `pkg-config`, but I don't think the problem is missing `pkg-config` itself, maybe missing `*.pc` file? Or a different `lzodefs.h` path.

**Comment by ygrek** [comment]
> So far, only a few distributions failed:

it is after pkg-config fix

> Yes, it depends on the pkg-config, but I don't think the problem is missing pkg-config itself, maybe missing *.pc file? Or a different lzodefs.h path.

.pc file is included with liblzo2-dev

**Comment by shonfeder** [comment]
I'm going to mark this as needing reporter action, since we are still getting failures:

On Arch

```
#=== ERROR while compiling lzo.0.0.3 ==========================================#
# context              2.2.0~rc1 | linux/x86_64 | ocaml-base-compiler.4.14.2 | pinned(https://github.com/ygrek/ocaml-lzo/releases/download/v0.0.3/lzo-0.0.3.tbz)
# path                 ~/.opam/4.14/.opam-switch/build/lzo.0.0.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p lzo -j 255
# exit-code            1
# env-file             ~/.opam/log/lzo-7-701449.env
# output-file          ~/.opam/log/lzo-7-701449.out
### output ###
# File "src/dune", line 7, characters 9-18:
# 7 |   (names lzo_stubs)
#              ^^^^^^^^^
# (cd _build/default/src && /usr/sbin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -g -I /home/opam/.opam/4.14/lib/ocaml -o lzo_stubs.o -c lzo_stubs.c)
# lzo_stubs.c:2:10: fatal error: lzodefs.h: No such file or directory
#     2 | #include <lzodefs.h>
#       |          ^~~~~~~~~~~
# compilation terminated.
```

and also

``
#=== ERROR while compiling lzo.0.0.3 ==========================================#
# context              2.2.0~rc1 | linux/x86_32 | ocaml-base-compiler.4.14.2 | pinned(https://github.com/ygrek/ocaml-lzo/releases/download/v0.0.3/lzo-0.0.3.tbz)
# path                 ~/.opam/4.14/.opam-switch/build/lzo.0.0.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p lzo -j 255
# exit-code            1
# env-file             ~/.opam/log/lzo-7-3dc7d0.env
# output-file          ~/.opam/log/lzo-7-3dc7d0.out
### output ###
# File "src/dune", line 7, characters 9-18:
# 7 |   (names lzo_stubs)
#              ^^^^^^^^^
# (cd _build/default/src && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -I/usr/include/lzo -g -I /home/opam/.opam/4.14/lib/ocaml -o lzo_stubs.o -c lzo_stubs.c)
# In file included from lzo_stubs.c:2:
# /usr/include/lzo/lzodefs.h:1035:6: error: #error "unexpected configuration - check your compiler defines"
#  1035 | #    error "unexpected configuration - check your compiler defines"
#       |      ^~~~~
# /usr/include/lzo/lzodefs.h:1038:6: error: #error "unexpected configuration - check your compiler defines"
#  1038 | #    error "unexpected configuration - check your compiler defines"
#       |      ^~~~~
```

It's not clear what the expected followup is here, i.e., whether you want to try to fix it for these platforms, record an incompatibility, etc.

Please let us know if this label is error tho, and you need input from the repo maintainers tho!

**Comment by mseri** [comment]
I think for 32 bits is enough to mark it unavailable

File: packages/lzo/lzo.0.0.3/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  ["lzo2"] {os-family = "bsd"}
+  ["liblzo2-dev"] {os-family = "debian"}
+  ["lzo"] {os-distribution = "homebrew"}
+  ["lzo-dev"] {os-distribution = "alpine"}
+  ["lzo-devel"] {os-family = "suse" | os-family = "opensuse"}
+  ["lzo-devel"] {os-distribution = "centos" | os-distribution = "fedora"}
+  ["lzo"] {os-family = "archlinux"}
+  ["lzo-devel"] {os-distribution = "ol"}
+]
+synopsis: "Bindings to LZO - a portable lossless data compression library"
```
Comment (mseri) [inline packages/lzo/lzo.0.0.3/opam:29]:
```suggestion
synopsis: "Bindings to LZO - a portable lossless data compression library"
available: arch != "x86_32" & arch != "arm32"
```

**Comment by mseri** [comment]
Thanks

**Pull request merged**